pkg/db/db.go

```diff
@@ -25,12 +25,24 @@ func db() *sql.DB {
 
 // EnsureLatestVersion ensures that the database is at the latest version by running all migrations.
 func EnsureLatestVersion() {
-	err := goose.SetDialect("sqlite3")
+	driver := config.GetDBDriver()
```

Comment:
I have modified related files on my branch: [db.go](https://github.com/karmek-k/server/blob/feature/migrate-db-library-to-goqu/pkg/db/db.go), [main.go](https://github.com/karmek-k/server/blob/feature/migrate-db-library-to-goqu/cmd/mangatsu-server/main.go)

In particular:
- lines 47, 49 in `db.go`
- line 14 in `main.go`

Briefly, in my version the dialect and the connection string are injected to `Initdb` from `main.go`. I haven't handled the migrations path, but it could be added to the `Database` struct.

What do you think?

Reply:
> but it could be added to the Database struct.

Sounds good to me.

Reply:
Done, here's my version: https://github.com/karmek-k/server/commit/8d08b6f52490052ace2586729aa6775b9366548c

Reply:
Looks good. I was wondering if we should do the dialect and path selecting maybe in the same function. Like this
```go
func GetDialectAndMigrationsPath(driver string) (string, string) {
	switch driver {
	case config.SQLite:
		return "sqlite3", "./pkg/db/migrations/sqlite"
	case config.PostgreSQL:
		return "postgres", "./pkg/db/migrations/psql"
	default:
		log.Fatal("Invalid DB driver", "driver", driver)
		return "", ""
	}
}```

Reply:
The code should never go into the default state though as [GetDBDriver](https://github.com/Mangatsu/server/blob/66faa624c89f0b7b727037ec9217a6d169ee1f37/internal/config/db.go#L11) ensures it's always sqlite, postgres or mysql.

Reply:
It's a good idea, but `(string, string)` should be changed to a concrete struct ([primitive obsession](https://refactoring.guru/smells/primitive-obsession)).

After that, we can save this struct to the `Database` wrapper.

> The code should never go into the default state though as [GetDBDriver](https://github.com/Mangatsu/server/blob/66faa624c89f0b7b727037ec9217a6d169ee1f37/internal/config/db.go#L11) ensures it's always sqlite, postgres or mysql.

It's just that the compiler complains if there is no `default` block with a `return` statement.

By the way, I'm somehow tired now, I'll continue tomorrow after work. See ya!

Reply:
Actually scratch that. I pushed new a [commit](https://github.com/Mangatsu/server/pull/18/files#diff-d7feb61ec3128e0d4ad454e4bc747d98a229f0cc71061f9bddb43f0ec8a0b70dR20) where I'm selecting them both directly in the config package. How does this look? @karmek-k

Reply:
> It's a good idea, but (string, string) should be changed to a concrete struct ([primitive obsession](https://refactoring.guru/smells/primitive-obsession)).

Got it.

> By the way, I'm somehow tired now, I'll continue tomorrow after work. See ya!

No problem and thanks a lot for the work you've already done!